http/logical.go

```diff
@@ -141,7 +141,6 @@ func buildLogicalRequestNoAuth(perfStandby bool, w http.ResponseWriter, r *http.
 		}
 
 	case "OPTIONS":
-	case "HEAD":
```

Comment:
This change seems unrelated to the resource quotas changes. Do you remember why it was added?

Reply:
I remember adding it as part of quotas changes. It was needed to be able to call buildLogicalRequestNoAuth for all the requests. I don't remember the exact failure that had triggered this (it was from tests though). Now that we reverted these changes, I figured there is no reason to keep this unjustified change, and reverted this as well.